R/redcap_import_select.R

```diff
@@ -0,0 +1,313 @@
+
+    cat("This script will loop you through the variable names in the provided data table and compares them with the names in a REDCap data dictionary.\n\n")
+    if (auto_match) {
+      cat("Auto-selecting of matching variables has been turned on! (To turn it off, set 'auto_match = FALSE')\n")
+      cat("If a matching variable name is found in the REDCap dictionary, the variable will be automatically selected without renaming.\n\n")
+    } else {
+      cat("Auto-selecting of matching variables has been turned off! (To turn it on set 'auto_match = TRUE')\n")
+      cat("If a matching variable name is found in the REDCap dictionary, you can decide to not select this variable at all or to select and rename it.\n\n")
+    }
+    if (auto_skip_nomatch) {
+      cat("Auto-skipping of non-matching variables has been turned on! (To turn it off, set 'auto_skip_nomatch = FALSE')\n")
+      cat("If a variable name is found with no matching name in the REDCap dictionary, the variable will be automatically skipped.\n\n")
+    } else {
+      cat("Auto-selecting of non-matching variables has been turned off! (To turn it on set 'auto_skip_nomatch = TRUE')\n")
+      cat("If a matching variable name is found with no matching name in the REDCap dictionary, you can decide to not select this variable at all or to select and rename it.\n\n")
+    }
+
+    cat(bold("PLEASE BE CAREFUL WITH YOUR CHOICES AS IT IS NOT POSSIBLE TO GO BACK IN THE LOOP!!\n\n"))
+
+    cat("You can press 'Esc' any time to stop the function but it is advised to finish the loop properly! This can be done by either typing 'exit' in the prompt or by looping through all the variables. It makes sure that the executed code is properly written in the log-file and can be copy-pasted into your R-script and adjusted manually at a later time.\n\n")
+
+    if(is.null(dict)) {
+      cat("No data dictionary has been provided!\n")
+      cat("The dictionary will be downloaded from REDCap with the URL and token you have provided.\n\n")
+    } else {
+      cat("A data dictionary has been provided.\n")
+      cat("Variable names will be read from this dictionary.\n\n")
+    }
+
+    if(!is.null(forms)) {
+      cat("Form selection has been restricted!\n")
+      cat("Only variable names from the following forms will be displayed:\n ")
+      cat(paste0(forms, sep="\n"))
+      cat("\n\n")
+    } else {
+      cat("No forms have been provided!\n")
+      cat("Variable names of all forms will be displayed.\n\n\n")
+    }
+
+    cat("Are you ready to begin? \n 1 = YES\n'esc' = STOP")
+    ans <- ""
+    while (ans != 1) {
+      ans <- readline(prompt="Answer= ")
+      if (ans != 1) {
+        cat("Please check your answer! \n 1 = YES\n'esc' = STOP")
+      }
+    }
+
+    cat("\nGreat! Let's begin!\n")
+    cat("\n-----------------------------------------------------------------\n\n")
+    Sys.sleep(wait)
+  }
+
+
+  # open log-files
+
+  if(log) {
+    log_file <- "redcap_import_select_code.txt"
```

Comment:
same here, maybe give the user control of the log file name?

Reply:
the log-file is appended each time the code is run. but it still seems useful to let the user decide where and under what name to save it. i implemented this option